volatility3/framework/contexts/__init__.py

```diff
@@ -462,3 +462,149 @@ def __init__(
         Module.__init__(
             self, context, name, layer_name, offset, symbol_table_name, layer_name
         )
+
+
+class MacOSKernelCacheSupportModule(Module):
+    """
+    If the MH_FILESET KernelCache support is ON, header addresses in all mach-o segments and sections
+    of the MH_FILESET are slid by a specific offset. This is problematic, as some kernel symbols, typically
+    contained in an external ISF file, won't be correctly readable with a sole KASLR shift.
+    The original "kernel" module is the "KernelCache" module, but for compatibility reasons
+    it is still named "kernel".
+    To circumvent this, we create an additional module object, in this context, with the
+    "vm_kernel_slide" shift as offset. Doing so, we are able to detect which slide to use,
+    depending on a provided symbol address.
+
+    Additional reference, on the stage where the KernelCache is given an additional slide :
+     - https://github.com/apple-open-source/macos/blob/14.3/xnu/osfmk/i386/i386_init.c#L621
+    """
+
+    def __init__(
+        self, context: interfaces.context.ContextInterface, config_path: str, name: str
+    ) -> None:
+        super().__init__(context, config_path, name)
+
+        pathjoin = interfaces.configuration.path_join
+        context.config[pathjoin(config_path, "vm_kernel_slide")] = context.config[
+            pathjoin(config_path, self.layer_name, "vm_kernel_slide")
+        ]
+        context.config[pathjoin(config_path, "kernel_start")] = (
```

Comment:
`offset` (a.k.a `kernel_virtual_offset`) is different from `vm_kernel_slide` (in the case of an MH_FILESET KernelCache), and needs to be passed as a config option.

Reply:
`kernel_virtual_offset` is what we've typically passed in to the kernel module, but *all* modules, regardless of which module, have the common values of `offset` and `size`, but for the new module you've constructed you've chosen different names.  When I add those into the `get_requirements` of the `Module`/`ModuleInterface` you'll see that you're defining the same values but with a different name.  I'd prefer we stick with the generic value names that modules already have rather than custom ones for this specific type of module.

Reply:
I consider `kernel_start` and `kernel_end` as additional requirements for this specific module.